src/node.cpp

```diff
@@ -1597,7 +1588,7 @@ LocalNode::~LocalNode()
         // shutting down
         if (sync->state < SYNC_INITIALSCAN)
         {
-            node->localnode = NULL;
+            node.reset();
```

Comment:
This wouldn't be required as it'll happen upon crossref_ptr dtor, wouldn't it?

Reply:
Yes - though, for a hotfix I think it's good to keep the changes to a minimum and have a clear one-to-one mapping from old to new code.   Though, we could reduce the last 15 lines of this function to just 4.  Should we make the change for this hotfix though?

Reply:
And it's interesting that the other case did not clear those pointers - that's one possible case where the problem could have occurred.   Now if we haven't called reset() ourselves that other case will be covered.

Reply:
yep, that was what I was looking out when I added the comment :-), but MegaClient::movetosyncdebris did set the localnode to NULL.
We can keep the code as is. worst case scenario: we will wonder why in a few months when we forget about this PR ;-P